src/irmin-pack/content_addressable_intf.ml

```diff
@@ -25,64 +25,80 @@ module type S = sig
   val unsafe_find : check_integrity:bool -> 'a t -> key -> value option
-  val flush : ?index:bool -> ?index_merge:bool -> 'a t -> unit
 
-  val sync : ?on_generation_change:(unit -> unit) -> 'a t -> unit
-  (** syncs a readonly instance with the files on disk. The same file instance
-      is shared between several pack instances. Therefore only the first pack
-      instance that checks a generation change, can see it.
-      [on_generation_change] is a callback for all pack instances to react to a
-      generation change. *)
-
-  val version : 'a t -> Version.t
   val generation : 'a t -> int63
-  val offset : 'a t -> int63
-
-  (** @inline *)
-  include S.Checkable with type 'a t := 'a t and type key := key
-
-  val clear_caches : 'a t -> unit
-  (** [clear_cache t] clears all the in-memory caches of [t]. Persistent data
-      are not removed. *)
+  (** The number of times that {!clear} has been called on this store. *)
 
   val clear_keep_generation : 'a t -> unit Lwt.t
 end
 
 module type Maker = sig
   type key
-  type index
 
   (** Save multiple kind of values in the same pack file. Values will be
       distinguished using [V.kind], so they have to all be different. *)
   module Make (V : Pack_value.S with type hash := key) :
-    S with type key = key and type value = V.t and type index = index
+    S with type key = key and type value = V.t
+end
+
+(** {!Memory} is {!S} extended with a basic [unit -> t] constructor. *)
+module Memory = struct
+  module type S = sig
+    include S
+
+    val v : string -> read t Lwt.t
+  end
+
+  module type Maker = sig
+    type key
+
+    module Make (V : Pack_value.S with type hash := key) :
+      S with type key = key and type value = V.t
+  end
+end
+
+(** {!Persistent} is {!S} extended with a constructor for *)
+module Persistent = struct
+  module type S = sig
+    include S
+
+    type index
+
+    val v :
+      ?fresh:bool ->
+      ?readonly:bool ->
+      ?lru_size:int ->
+      index:index ->
+      string ->
+      read t Lwt.t
+  end
+
+  module type Maker = sig
+    type key
+    type index
+
+    (** Save multiple kind of values in the same pack file. Values will be
+        distinguished using [V.magic], so they have to all be different. *)
+    module Make (V : Pack_value.S with type hash := key) :
+      S with type key = key and type value = V.t and type index = index
+  end
 end
 
 module type Sigs = sig
   module type S = S
-  module type Maker = Maker
 
-  module Maker
-      (_ : Version.S)
-      (Index : Pack_index.S)
-      (K : Irmin.Hash.S with type t = Index.key) :
-    Maker with type key = K.t and type index = Index.t
+  module Closeable (CA : S) : sig
+    include S with type key = CA.key and type value = CA.value
+
+    val make_closeable : 'a CA.t -> 'a t
+    val get_open_exn : 'a t -> 'a CA.t
 
-  module Closeable (CA : S) :
-    S with type key = CA.key and type value = CA.value and type index = CA.index
+    val unsafe_get_inner_store : 'a t -> 'a CA.t
```

Comment:
I think you are not using this anywhere.

Reply:
Indeed; nice spot.